components/apimgt/org.wso2.carbon.apimgt.gateway/src/test/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessorTest.java

```diff
@@ -315,6 +322,59 @@ public void testHandleRequestInvalidScope() throws Exception  {
         Assert.assertFalse(processorResponseDTO.isCloseConnection());
     }
 
+    @Test
+    public void testHandleRequestScopeValidationSkipWhenSecurityDisabled() throws Exception  {
+
+        InboundMessageContext inboundMessageContext = new InboundMessageContext();
+        int msgSize = 100;
+        String msgText = "{\"id\":\"1\",\"type\":\"start\",\"payload\":{\"variables\":{},\"extensions\":{},"
+                + "\"operationName\":null,\"query\":\"subscription {\\n  "
+                + "liftStatusChange {\\n    id\\n    name\\n }\\n}\\n\"}}";
+        PowerMockito.mockStatic(InboundWebsocketProcessorUtil.class);
+        InboundProcessorResponseDTO responseDTO = new InboundProcessorResponseDTO();
+        PowerMockito.when(InboundWebsocketProcessorUtil.authenticateToken(inboundMessageContext))
+                .thenReturn(responseDTO);
+
+        // Get schema and parse
+        String graphqlDirPath = "graphQL" + File.separator;
+        String relativePath = graphqlDirPath + "schema_with_additional_props.graphql";
+        String schemaString = IOUtils.toString(getClass().getClassLoader().getResourceAsStream(relativePath));
+        SchemaParser schemaParser = new SchemaParser();
+        TypeDefinitionRegistry registry = schemaParser.parse(schemaString);
+        GraphQLSchema schema = UnExecutableSchemaGenerator.makeUnExecutableSchema(registry);
+        GraphQLSchemaDTO schemaDTO = new GraphQLSchemaDTO(schema, registry);
+        inboundMessageContext.setGraphQLSchemaDTO(schemaDTO);
+
+        // VerbInfoDTO with security disabled
+        VerbInfoDTO verbInfoDTO = new VerbInfoDTO();
+        verbInfoDTO.setHttpVerb("SUBSCRIPTION");
+        verbInfoDTO.setThrottling("Unlimited");
+        verbInfoDTO.setAuthType("None");
+        PowerMockito.when(InboundWebsocketProcessorUtil.findMatchingVerb("liftStatusChange",
+                inboundMessageContext)).thenReturn(verbInfoDTO);
+
+        // Creating response for scope validation
+        GraphQLProcessorResponseDTO graphQLProcessorResponseDTO = new GraphQLProcessorResponseDTO();
+        graphQLProcessorResponseDTO.setError(true);
+        graphQLProcessorResponseDTO.setErrorCode(WebSocketApiConstants.FrameErrorConstants.RESOURCE_FORBIDDEN_ERROR);
+        graphQLProcessorResponseDTO.setErrorMessage("User is NOT authorized to access the Resource");
+        graphQLProcessorResponseDTO.setCloseConnection(false);
```

Comment:
in this scenario, shouldn't this be a successful scenario instead of 401 error?

Reply:
this is okay. noticed we are checking if a 401 scope validation error is not returned when auth type is none.